cairo1-run/src/main.rs

```diff
@@ -358,6 +365,203 @@ fn main() -> Result<(), Error> {
+        // and `pc` registers.
+        ret;
+    }
+    .instructions
+}
+
+/// Returns the instructions to add to the beginning of the code to successfully call the main
+/// function, as well as the builtins required to execute the program.
+pub fn create_entry_code(
+    sierra_program_registry: &ProgramRegistry<CoreType, CoreLibfunc>,
+    casm_program: &CairoProgram,
+    type_sizes: &UnorderedHashMap<ConcreteTypeId, i16>,
+    func: &Function,
+    initial_gas: usize,
+) -> Result<(Vec<Instruction>, Vec<BuiltinName>), RunnerError> {
+    let mut ctx = casm! {};
+    // The builtins in the formatting expected by the runner.
+    let builtins = vec![
+        BuiltinName::pedersen,
+        BuiltinName::range_check,
+        BuiltinName::bitwise,
+        BuiltinName::ec_op,
+        BuiltinName::poseidon,
+    ];
+    // The offset [fp - i] for each of this builtins in this configuration.
+    let builtin_offset: HashMap<cairo_lang_sierra::ids::GenericTypeId, i16> = HashMap::from([
+        (PedersenType::ID, 7),
+        (RangeCheckType::ID, 6),
+        (BitwiseType::ID, 5),
+        (EcOpType::ID, 4),
+        (PoseidonType::ID, 3),
+    ]);
+    // Load all vecs to memory.
+    let mut ap_offset: i16 = 0;
+    let after_vecs_offset = ap_offset;
+    if func
+        .signature
+        .param_types
+        .iter()
+        .any(|ty| get_info(sierra_program_registry, ty).long_id.generic_id == SegmentArenaType::ID)
+    {
+        casm_extend! {ctx,
+            // SegmentArena segment.
+            %{ memory[ap + 0] = segments.add() %}
+            // Infos segment.
+            %{ memory[ap + 1] = segments.add() %}
+            ap += 2;
+            [ap + 0] = 0, ap++;
+            // Write Infos segment, n_constructed (0), and n_destructed (0) to the segment.
+            [ap - 2] = [[ap - 3]];
+            [ap - 1] = [[ap - 3] + 1];
+            [ap - 1] = [[ap - 3] + 2];
+        }
+        ap_offset += 3;
+    }
+    for ty in func.signature.param_types.iter() {
+        let info = get_info(sierra_program_registry, ty);
+        let ty_size = type_sizes[ty];
+        let generic_ty = &info.long_id.generic_id;
+        if let Some(offset) = builtin_offset.get(generic_ty) {
+            casm_extend! {ctx,
+                [ap + 0] = [fp - offset], ap++;
+            }
+        } else if generic_ty == &SystemType::ID {
+            casm_extend! {ctx,
+                %{ memory[ap + 0] = segments.add() %}
+                ap += 1;
+            }
+        } else if generic_ty == &GasBuiltinType::ID {
+            casm_extend! {ctx,
+                [ap + 0] = initial_gas, ap++;
+            }
+        } else if generic_ty == &SegmentArenaType::ID {
+            let offset = -ap_offset + after_vecs_offset;
+            casm_extend! {ctx,
+                [ap + 0] = [ap + offset] + 3, ap++;
+            }
+            // } else if let Some(Arg::Array(_)) = arg_iter.peek() {
```

Comment:
Are we leaving this commented for a reason?

Reply:
Its the part that handles function arguments (which we are not using), I left it for future reference

Reply:
great!